File: practice/35_Search_Insert_Position.kt
fun main() {
	println(searchInsert(intArrayOf(1, 3, 5, 6), 8))
}

fun searchInsert(nums: IntArray, target: Int): Int {
	var start=0
	var end = nums.size-1

	while(start<=end) {
		var mid = start+(end-start)/2
		if(nums[mid] == target) {
			return mid
		} else if(nums[mid] < target) {
			start = mid+1
		} else {
			end = mid-1
		}
	}

	return start
}

File: practice/35_Search_Insert_Position.js
function searchInsert(nums, target) {
    let start = 0,
        end = nums.length - 1,
        result = 0;

    while (start <= end) {
        let mid = Math.floor(start + (end - start) / 2);
        if (nums[mid] === target) {
            return mid;
        } else if (nums[mid] < target) {
            result = mid + 1;
            start = mid + 1;
        } else {
            result = mid;
            end = mid - 1;
        }
    }

    return result;
}

// console.log(searchInsert([1, 3, 5, 6], 5));
console.log(searchInsert([1, 3, 5, 6], 2));
console.log(searchInsert([1, 3, 5, 6], 7));
